### ISSUE TITLE
HeadRowColor will over table decoration color,and can not set borderRadius. like image.

### ISSUE THREAD

**Issue by guuguo** [maxim-saplin/data_table_2#194]
I set the background color for the table and the color for the headerRow, but the color covered the background and truncated the border.As shown in the picture, take a look at the strange style of the corners. I hope Headrow can also set rounded corners。
<img width="1179" alt="image" src="https://github.com/maxim-saplin/data_table_2/assets/12984352/392f912e-0143-4a5e-ac11-5c138b4707de">
this is my code:
```
DataTable2(
                headingRowHeight: 50,
                headingRowColor:
                    MaterialStateProperty.all(colorScheme(context).surfaceDim),
                border: TableBorder(
                  borderRadius: BorderRadius.only(topLeft: Radius.circular(20),topRight: Radius.circular(20)),
                  // bottom:  BorderSide(color: colorScheme(context).outlineVariant),
                  horizontalInside: BorderSide(color: colorScheme(context).outlineVariant,width: 2),
                  // verticalInside:  BorderSide(color: colorScheme(context).outlineVariant,width: 2),
                ),
                decoration: BoxDecoration(
                  border: Border.all(color: colorScheme(context).outlineVariant),
                  color: colorScheme(context).surface,
                  borderRadius: BorderRadius.circular(20),
                ))
```

**Comment by maxim-saplin** [comment]
Thanks, will check

**Comment by maxim-saplin** [comment]
As far as I see the problem, box decoration doesn't get applied to header while working fine with bottom side:
![image](https://github.com/maxim-saplin/data_table_2/assets/7947027/b4705ae7-14fa-499a-9023-b77fa60320d3)

**Comment by maxim-saplin** [comment]
I've published 2.5.3 to pub.dev, added clipBehavior param to constructor, try setting it anything different from None
